lib/datadog/tracing/contrib/redis/patcher.rb

```diff
@@ -21,9 +21,7 @@ def self.included(base)
             # Instance method patch for redis instance
             module InstanceMethods
               def initialize(options = {})
-                options[:redis_instance] = self
-
-                super(options)
+                super(options.merge(redis_instance: self))
               end
```

Comment:
Minor: It may be worth calling out specifically in a comment that `options` can be frozen and that this has been an issue for customers in the past. (So that the next refactoring we do here doesn't accidentally reintroduce the issue)